### DISCUSSION TITLE
Brew uninstall --zap does not remove any files

### DISCUSSION THREAD

**Discussion by SoCuul** [Homebrew/discussions#5740] [Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.4.5
ORIGIN: https://github.com/Homebrew/brew
HEAD: 254bf3fe9d8fa2e1b2fb55dbcf535b2d870180c4
Last commit: 6 days ago
Core tap HEAD: 6d5d6f334373cfa00e21acab45a418144b537c15
Core tap last commit: 89 minutes ago
Core tap JSON: 12 Nov 03:27 UTC
Core cask tap HEAD: cafc37a60ab2c77820e20131e9a6c1465aef3128
Core cask tap last commit: 37 minutes ago
Core cask tap JSON: 12 Nov 03:27 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.qwcsc55102/org.xquartz:0
HOMEBREW_MAKE_JOBS: 16
HOMEBREW_NO_AUTO_UPDATE: set
HOMEBREW_NO_ENV_HINTS: set
HOMEBREW_NO_INSTALL_FROM_API: set
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.3.6 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.6/bin/ruby
CPU: 16-core 64-bit dunno
Clang: 16.0.0 build 1600
Git: 2.47.0 => /opt/homebrew/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.1-arm64
CLT: 16.0.0.26.4
Xcode: 16.1
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  cowsay

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libfuse-t-1.0.44.dylib

Warning: Unbrewed '.pc' files were found in /usr/local/lib/pkgconfig.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected '.pc' files:
  /usr/local/lib/pkgconfig/fuse-t.pc

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libfuse-t-1.0.44.a
```

### Description of issue

Whenever I run `brew uninstall --zap <app>` the configuration files are not removed.

The files that should be removed are displayed in the CLI, but after the uninstall is finished, none of the files mentioned have been touched.

### Relevant casks

All

**Comment by gromgit** [comment]
Generalizations like "Relevant casks: All" aren't useful input for debugging (and quite likely wrong). Can you show us the output of a _actual_ `brew uninstall --zap <cask>`, during which you noticed the files not being removed?

**Reply by SoCuul** [reply]
> Generalizations like "Relevant casks: All" aren't useful input for debugging (and quite likely wrong). Can you show us the output of a _specific_ zap session, during which you noticed the files not being removed?

I used this generalization as I tried it with several casks including gimp and inkscape.

https://github.com/user-attachments/assets/dad657bd-053b-4e50-be12-e190d4bd90de

**Reply by gromgit** [reply]
What's the output of the following _in text please_ (not a screen capture):
```
cd ~/Library/Preferences
ls -ld . org.inkscape.Inkscape.plist org.gimp.gimp-2.0:.plist
```
?

**Reply by SoCuul** [reply]
```sh
$ cd ~/Library/Preferences
$ ls -ld . org.inkscape.Inkscape.plist org.gimp.gimp-2.0:.plist

ls: org.gimp.gimp-2.0:.plist: No such file or directory
drwx------+ 1078 daniel  staff  34496 Nov 17 06:33 .
-rw-------@    1 daniel  staff    793 Dec 31  2023 org.inkscape.Inkscape.plist

$ ls -ld org.gimp.gimp-2.10.plist

-rw-------@ 1 daniel  staff  126 Jul 22  2023 org.gimp.gimp-2.10.plist

**Reply by gromgit** [reply]
Note that the colon in the `gimp` plist filename I wrote is deliberate. If you look carefully at the `brew zap` output, that's the exact filename. This seems to be a bug in the cask file.

In any case, what's the output of:
```
id
cd ~/Library/Preferences
/bin/ls -lde@ . org.inkscape.Inkscape.plist org.gimp.gimp-2.0.plist
```
?

**Reply by SoCuul** [reply]
Yes I know, I purposely provided an output with and without the colon in my previous response for this exact reason.

**Reply by SoCuul** [reply]
```sh
$ /bin/ls -lde@ . org.inkscape.Inkscape.plist org.gimp.gimp-2.10.plist

drwx------+ 1078 daniel  staff  34496 Nov 17 06:50 .
 0: group:everyone deny delete
-rw-------@    1 daniel  staff    126 Jul 22  2023 org.gimp.gimp-2.10.plist
	com.apple.lastuseddate#PS	   16
	com.apple.provenance	   11
-rw-------@    1 daniel  staff    793 Dec 31  2023 org.inkscape.Inkscape.plist
	com.apple.macl	   72
```

**Reply by SoCuul** [reply]
Additionally, this is the PR where the zap for the plist was added: https://github.com/Homebrew/homebrew-cask/commit/a433a85dda5b6d55a335b5306d7bafa807a7ab98

> At some point between gimp 2.8 and 2.10 they changed their application domain from `org.gnome.gimp` to `org.gimp.gimp`. Also, even if a little bit strange, the `.plist` and `.savedState` files have a colon/slash after the version number.

**Reply by gromgit** [reply]
I [submitted a PR](https://github.com/Homebrew/homebrew-cask/pull/192359#pullrequestreview-2441051917) to fix the GIMP paths, and it's been merged. `brew update`, then reinstall GIMP and see if an uninstall now correctly trashes the indicated paths.

**Reply by SoCuul** [reply]
> I [submitted a PR](https://github.com/Homebrew/homebrew-cask/pull/192359#pullrequestreview-2441051917) to fix the GIMP paths, and it's been merged. `brew update`, then reinstall GIMP and see if an uninstall now correctly trashes the indicated paths.

Well considering inkscape had the correct path the whole time and didn't work, nothing will have changed.

**Reply by gromgit** [reply]
Maybe, maybe not. You're already assuming both failed for the same reason on your system, when the GIMP cask had a clear issue that affects _everyone_.

Do you have a specific reason not to try?
